rca/programmes/models.py

```diff
@@ -917,6 +926,54 @@ def listing_meta(self):
         # Returns a page 'type' value that's readable for listings,
         return "Programme"
 
+    def get_what_you_will_cover_blocks_accordion_snippet(self):
+        """
+        We cannot directly index the accordion_snippet block in the what_you_will_cover_blocks
+        StreamField. We therefore have to do this manually.
+
+        Ref: https://docs.wagtail.org/en/stable/topics/search/indexing.html#indexing-callables-and-other-attributes
+        """
+
+        if not self.what_you_will_cover_blocks:
+            return None
+
+        data = self.what_you_will_cover_blocks.get_prep_value()
+        accordion_snippet_values = [
+            item["value"] for item in data if item["type"] == "accordion_snippet"
```

Comment:
Given the None return on 945, is it worth using `item.get("type")` to avoid potential key errors if "type" is missing in any dictionary here?

Reply:
updated in b4d80600